static/js/tools/stat_var/dataset_selector.tsx

```diff
@@ -51,31 +51,87 @@ export function DatasetSelector(props: DatasetSelectorProps): JSX.Element {
   const [showSuggestions, setShowSuggestions] = useState(false);
   const [userInput, setUserInput] = useState("");
 
+  function updateSource(source: string): void {
+    if (!source) {
+      setUserInput("");
+      return;
+    }
+    axios
+      .get(`/api/stats/propvals/name/${source}`)
+      .then((resp) => {
+        setUserInput(
+          resp.data[source].length > 0 ? resp.data[source][0] : source
+        );
+      })
+      .catch(() => {
+        setUserInput("");
+      });
+  }
+
   function updateDatasets(source: string): void {
-    if (!(source in props.sourceMap)) {
+    if (!source) {
+      setDatasets([]);
       return;
     }
-    const dcid = props.sourceMap[source];
     axios
-      .get(`/api/browser/propvals/isPartOf/${dcid}`)
+      .get(`/api/browser/propvals/isPartOf/${source}`)
       .then((resp) => {
         const currentDatasets = [];
+        const datasetSet = new Set();
         for (const dataset of resp.data?.values?.in) {
+          // Remove duplicates.
+          if (datasetSet.has(dataset.dcid)) {
+            continue;
+          }
           currentDatasets.push({
             dcid: dataset.dcid,
             name: dataset.name,
           });
+          datasetSet.add(dataset.dcid);
         }
         currentDatasets.sort((a, b): number => {
           return a.name.localeCompare(b.name);
         });
         setDatasets(currentDatasets);
+        const dataset = getUrlToken(DATASET_PARAM);
+        if (!dataset || !currentDatasets.some((d) => d.dcid === dataset)) {
+          return;
+        }
+        const datasetOption = document.getElementById(
+          `${CSS_PREFIX}-${dataset}`
+        );
+        datasetOption.setAttribute("selected", "true");
       })
       .catch(() => {
         setDatasets([]);
       });
   }
 
+  useEffect(() => {
+    const handleHashChange = () => {
```

Comment:
I think it gets a little confusing if hashChange is handled in multiple places. It'll be better to handle it in page.tsx and pass along the information as props to dataset_selector. So probably want DatasetSelector to take in the selected source and dataset

Reply:
done